tests/api/db/test_artifacts.py

```diff
@@ -853,6 +854,83 @@ def test_list_artifact_for_tagging_fallback(
         )
         assert len(artifacts) == 1
 
+    def test_iterations_with_latest_tag(self, db: DBInterface, db_session: Session):
+        project = "artifact_project"
+        artifact_key = "artifact_key"
+        artifact_tree = "artifact_tree"
+        artifact_body = self._generate_artifact(
+            artifact_key, tree=artifact_tree, project=project
+        )
+        iteration_number = 5
+
+        # create artifacts with the same key and different iterations
+        for iteration in range(1, iteration_number + 1):
+            artifact_body["metadata"]["iter"] = iteration
+            db.store_artifact(
+                db_session,
+                artifact_key,
+                artifact_body,
+                project=project,
+                iter=iteration,
+                producer_id=artifact_tree,
+            )
+
+        # list artifact with "latest" tag - should return all artifacts
+        artifacts = db.list_artifacts(db_session, project=project, tag="latest")
+        assert len(artifacts) == iteration_number
+
+        # mark iteration 3 as the best iteration
+        best_iteration = 3
+        self._mark_best_iteration_artifact(
+            db, db_session, project, artifact_key, artifact_tree, best_iteration
+        )
```

Comment:
Instead of writing an extra function, you could have already marked that in the `store_artifact` above (with the `best_iteration` param), or did you want to specifically test the usage of link artifacts for that?

Reply:
@theSaarco I wanted to both test the link artifact functionality and to make sure the tagging isn't affected by the iterations / best iteration.